### PR TITLE
updates to fix warnings for cleaner plans

### PR DESCRIPTION
**Pull request by bvkin**
These updates will fix all warnings which appeared as a result of the terraform/aws provider updates

### CODE REVIEW

File: modules/vpc-endpoints/terraform.tf [hunk 1]
```diff
@@ -0,0 +1,7 @@
+terraform {
```
Comment (bvkin) [inline modules/vpc-endpoints/terraform.tf:1]: fixes:
> │ Warning: Reference to undefined provider
│ 
│   on main.tf line 256, in module "vpc_endpoints_use1":
│  256:     aws = aws.use1

File: main.tf [hunk 2]
```diff
@@ -442,19 +442,23 @@ module "ecs_cluster_asg" {
 resource "aws_s3_bucket" "binaries" {
   bucket        = "${data.aws_caller_identity.current.account_id}-${data.aws_region.current.name}-binaries"
   force_destroy = true
+  tags = local.tags
+}
 
-  server_side_encryption_configuration {
```
Comment (bvkin) [inline main.tf:446]: Fixes:
> │ Warning: Argument is deprecated
│ 
│   with aws_s3_bucket.binaries,
│   on main.tf line 442, in resource "aws_s3_bucket" "binaries":
│  442: resource "aws_s3_bucket" "binaries" {

File: modules/test-controller-deploy/main.tf [hunk 3]
```diff
@@ -538,6 +539,11 @@ BUILDSPEC
   tags = local.tags
 }
 
+resource "aws_codestarconnections_connection" "this" {
```
Comment (bvkin) [inline modules/test-controller-deploy/main.tf]: Will require a login to codepipeline from the console
Fixes
> Fixes
Warning: The CodePipeline GitHub version 1 action provider is deprecated.

File: outputs.tf [hunk 4]
```diff
@@ -82,6 +82,6 @@ output "route53_endpoints" {
         bastion             = module.bastion.bastion_enpoint
         ui_endpoint         = module.test_controller_service.ui_endpoint
         agent_endpoint      = module.test_controller_service.agent_endpoint
-        opensearch_endpoint = module.opensearch[0].opensearch_endpoint
+        opensearch_endpoint = var.create_opensearch ? module.opensearch[0].opensearch_endpoint : ""
```
Comment (bvkin) [inline outputs.tf:85]: Allows a plan in an un-initialized environment to finish

**Review by jallen-frb — APPROVED** [review]

**Review by jallen-frb — APPROVED** [review]

**Review by kylecrawshaw — CHANGES_REQUESTED** [review]
Please sign commits to pass DCO.

**Review by kylecrawshaw — APPROVED** [review]
LGTM

**Pull request merged**